lib/manager/cargo/artifacts.js

```diff
@@ -1,27 +1,117 @@
+const upath = require('upath');
+const process = require('process');
+const fs = require('fs-extra');
+const { exec } = require('child-process-promise');
+
 module.exports = {
   getArtifacts,
 };
 
-async function getArtifacts(cargoTomlFileName) {
-  await logger.debug(`cargo.getArtifacts(${cargoTomlFileName})`);
-  let cargoLockFileName;
-  try {
-    cargoLockFileName = cargoTomlFileName.replace(/\.toml$/, '.lock');
-    logger.debug('Updating ' + cargoLockFileName);
-    // TODO: Update cargo lock file
+// TODO: Make docker binarySource work
+// TODO: Test with real crates
+async function getArtifacts(
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config
+) {
+  await logger.debug({ config }, `cargo.getArtifacts(${packageFileName})`);
+  if (updatedDeps === undefined || updatedDeps.length < 1) {
+    logger.debug('No updated cargo deps - returning null');
     return null;
-  } catch (err) {
+  }
+  if (!config.gitFs) {
     logger.warn(
-      { err, message: err.message },
-      'Failed to update Cargo lock file'
+      'Cargo lock updating requires gitFs to be enabled by the Renovate administrator.'
     );
+    return null;
+  }
+  const lockFileName = 'Cargo.lock';
+  const existingLockFileContent = await platform.getFile(lockFileName);
+  if (!existingLockFileContent) {
+    logger.debug('No Cargo.lock found');
+    return null;
+  }
+  const localPackageFileName = upath.join(config.localDir, packageFileName);
+  const localLockFileName = upath.join(config.localDir, lockFileName);
+  let stdout;
+  let stderr;
+  try {
+    await fs.outputFile(localPackageFileName, newPackageFileContent);
+    logger.debug('Updating ' + lockFileName);
+    for (let i = 0; i < updatedDeps.length; i += 1) {
+      const dep = updatedDeps[i];
+      // TODO: document this command and why we use it
```

Comment:
@nchashch could you add some documentation of these parameters and remove this TODO?

Reply:
Sure.

Reply:
I have added the documentation.